pytorch_lightning/core/optimizer.py

```diff
@@ -298,15 +302,21 @@ def _configure_schedulers_automatic_opt(schedulers: list, monitor: Optional[str]
             lr_schedulers.append(
                 {**default_config, "scheduler": scheduler, "reduce_on_plateau": True, "monitor": monitor}
             )
-        elif isinstance(scheduler, optim.lr_scheduler._LRScheduler):
-            lr_schedulers.append({**default_config, "scheduler": scheduler})
         else:
-            raise ValueError(f'The provided lr scheduler "{scheduler}" is invalid')
+            lr_schedulers.append({**default_config, "scheduler": scheduler})
+
+        current_scheduler = lr_schedulers[-1]["scheduler"]
+        if not isinstance(current_scheduler, _SupportsStateDict):
+            raise ValueError(f"The provided lr scheduler `{current_scheduler.__class__.__name__}` is invalid.")
```

Comment:
```suggestion
        if not isinstance(current_scheduler, _SupportsStateDict):
            raise ValueError(f"The provided lr scheduler `{current_scheduler.__class__.__name__}` is invalid.")
         if not isinstance(current_scheduler, _LR_Scheduler_Protocol) and is_overriden("lr_scheduler_step", self.lightning_module):
            raise ValueError(f"The provided lr scheduler doesn't follow the PyTorch LR Scheduler Protocol and you should override the `lr_scheduler_step` hook of the LightningModule with your own logic.")
```

Reply:
there is no `_LR_Scheduler_Protocol`, using the existing `_SupportsStateDict` since with custom hook, we don't force users to have the `step` method now.

Reply:
Hey @rohitgr7.

Yes, I meant we should implement a protocol for PyTorch LR Scheduler and if the user scheduler doesn't match and they haven't overridden the new hook, it should break.

Reply:
ah! good edge case 👍 thank you